File: python/bob/bob.py
def hey(phrase):    
    phrase = phrase.strip()

    if phrase == "":
        return "Fine. Be that way!"

    if phrase.endswith('?'):
        if phrase.isupper():
            return "Calm down, I know what I'm doing!"
        else:
            return "Sure."
    else:
        if phrase.isupper():
            return "Whoa, chill out!"
        else:
            return "Whatever."

File: javascript/bob/bob.js
export const hey = message => {
  const trimmed = message.trimStart().trimEnd();

  if (trimmed.length === 0) {
    return "Fine. Be that way!";
  }

  const isShouting =
    trimmed.toUpperCase() === trimmed && trimmed.toLowerCase() !== trimmed;

  const isQuestion = trimmed[trimmed.length - 1] === "?";

  if (isShouting && isQuestion) {
    return "Calm down, I know what I'm doing!";
  }

  if (isShouting) {
    return "Whoa, chill out!";
  }

  if (isQuestion) {
    return "Sure.";
  }

  return "Whatever.";
};
